lib/dentaku/ast/identifier.rb

```diff
@@ -12,6 +12,8 @@ def initialize(token)
       def value(context={})
         v = context[identifier]
```

Comment:
Maybe it would be better to see if the `context` has the key `identifier` here, rather than checking the to see if the returned value is `nil`.  That way, we could explicitly pass in `nil` as an identifier value.